src/utils/send-gtag-event.js

```diff
@@ -1,5 +1,7 @@
 export default function sendGtagEvent(eventName, properties) {
   if (window.zaraz) {
-    window.zaraz.track(eventName, properties);
+    const result = window.zaraz.track(eventName, properties);
+    return Promise.resolve(result);
```

Comment:
why do you wrap result in a promise?

Reply:
It was more to indicate that `sendGtagEvent` returns a promise so that, in the future, others will know to `await` it to avoid any potential race conditions. I could just add a comment instead, though.

Reply:
ideally it shouldn't be required to await, such things should be sent in batches, that's what segment does for example. Maybe we can use segment directly?

Reply:
Yup, we considered calling the endpoints directly (and I drafted this [one-pager](https://docs.google.com/document/d/1iDPiOB_aZK7b7jxyNr8j-nSnd7gvxwBwPxBCvm9t_so/edit?tab=t.0) here with the tradeoffs), but since this pipeline via Zaraz was already built out before, we decided to go with the quickest approach here. It also avoided a Databricks security review since we aren't adding any new API calls (even though the data still flows through segment). Here's the slack thread for more context: https://databricks.slack.com/archives/C091EQGTY67/p1754666579143419

Reply:
I don't think we need any security review for segment, we already use it. 
let's see if this satisfies the cause, but it looks pretty weird to me, that we are not using semgent directly, I don't see why, but it obviously causes troubles